$NetBSD: patch-otherlibs_Makefile,v 1.6 2017/07/17 23:26:23 wiz Exp $

Honor LDFLAGS.

Breaks dllunix.so on illumos since libnsl and libsocket are not linked due to overriding LDOPTS.

--- ocaml-4.06.0/otherlibs/Makefile.orig	2017-06-23 15:13:40.000000000 +0000
+++ ocaml-4.06.0/otherlibs/Makefile
@@ -53,6 +53,7 @@ MKLIB=$(CAMLRUN) $(ROOTDIR)/tools/ocamlm
 #LINKOPTS=
 #LDOPTS=
 #HEADERS=
+LDOPTS?=-ldopt "$(LDFLAGS)"
 
 CMIFILES ?= $(CAMLOBJS:.cmo=.cmi)
 CAMLOBJS_NAT ?= $(CAMLOBJS:.cmo=.cmx)
--- ocaml-4.06.0/otherlibs/unix/Makefile.orig	2020-05-28 18:06:09.681882256 +0000
+++ ocaml-4.06.0/otherlibs/unix/Makefile	2020-05-28 18:06:18.178471768 +0000
@@ -20,7 +20,7 @@
 EXTRACAMLFLAGS=-nolabels
 
 # dllunix.so particularly requires libm for modf symbols
-LDOPTS=$(NATIVECCLIBS)
+LDOPTS=-ldopt "-B direct $(NATIVECCLIBS)"
 
 COBJS=accept.o access.o addrofstr.o alarm.o bind.o chdir.o chmod.o \
   chown.o chroot.o close.o closedir.o connect.o cst2constr.o cstringv.o \
